File: library/src/main/java/com/lany/box/config/FragmentConfig.java
package com.lany.box.config;

import android.support.annotation.ColorRes;
import android.support.annotation.LayoutRes;

import lombok.Getter;

@Getter
public class FragmentConfig {
    /**
     * 布局文件id
     */
    @LayoutRes
    private int layoutId;
    /**
     * Toolbar布局文件id
     */
    @LayoutRes
    private int toolBarLayoutId;
    /**
     * 是否需要Toolbar
     */
    private boolean hasToolbar;
    /**
     * toolbar的高度
     */
    private int toolbarHeight;
    /**
     * Toolbar颜色
     */
    @ColorRes
    private int toolbarColor;
    /**
     * 内容的背景颜色
     */
    @ColorRes
    private int contentColor;

    public FragmentConfig layoutId(int layoutId) {
        this.layoutId = layoutId;
        return this;
    }

    public FragmentConfig toolBarLayoutId(int toolBarLayoutId) {
        this.toolBarLayoutId = toolBarLayoutId;
        this.hasToolbar = true;
        return this;
    }

    public FragmentConfig toolbarHeight(int toolbarHeight) {
        this.toolbarHeight = toolbarHeight;
        return this;
    }

    public FragmentConfig toolbarColor(int color) {
        this.toolbarColor = color;
        return this;
    }

    public FragmentConfig contentColor(@ColorRes int color) {
        this.contentColor = color;
        return this;
    }
}

File: library/src/main/java/com/github/lany192/box/config/FragmentConfig.kt
package com.github.lany192.box.config

import androidx.annotation.ColorRes
import androidx.annotation.LayoutRes

class FragmentConfig {
    /**
     * 布局文件id
     */
    @LayoutRes
    var layoutId = 0

    /**
     * Toolbar布局文件id
     */
    @LayoutRes
    var toolBarLayoutId = 0

    /**
     * 是否需要Toolbar
     */
    var hasToolbar = false

    /**
     * toolbar的高度
     */
    var toolbarHeight = 0

    /**
     * Toolbar颜色
     */
    @ColorRes
    var toolbarColor = 0

    /**
     * 内容的背景颜色
     */
    @ColorRes
    var contentColor = 0

    fun layoutId(layoutId: Int): FragmentConfig {
        this.layoutId = layoutId
        return this
    }

    fun toolBarLayoutId(toolBarLayoutId: Int): FragmentConfig {
        this.toolBarLayoutId = toolBarLayoutId
        hasToolbar = true
        return this
    }

    fun toolbarHeight(toolbarHeight: Int): FragmentConfig {
        this.toolbarHeight = toolbarHeight
        return this
    }

    fun toolbarColor(color: Int): FragmentConfig {
        toolbarColor = color
        return this
    }

    fun contentColor(@ColorRes color: Int): FragmentConfig {
        contentColor = color
        return this
    }
}